ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -124,6 +255,20 @@ public String getLongForm() {
     StringBuilder sb = new StringBuilder();
     sb.append("[").append(getID());
     sb.append(":").append(version);
+    switch (version) {
+      case BLOB_ID_V1:
+        break;
+
+      case BLOB_ID_V2:
+        sb.append(":").append(flag);
+        sb.append(":").append(datacenterId);
+        sb.append(":").append(accountId);
+        sb.append(":").append(containerId);
+        break;
+
+      default:
+        throw new IllegalArgumentException("blob id version=" + version + " not supported");
+    }
```

Comment:
a question about compareTo() methods that is untouched. Shouldn't we compare getID() and not individual partitionId and uuid? StoreKey comparison is for the purpose of index entries right.